pkgs/applications/misc/k2pdfopt/default.nix

```diff
@@ -11,44 +11,119 @@
 
 with stdenv.lib;
 
-stdenv.mkDerivation rec {
-  pname = "k2pdfopt";
-  version = "2.51a";
+# k2pdfopt is a pain to package. Instead of shipping patches for mupdf,
+# leptonica, and tessertact, k2pdfopt includes modified source files for these
+# packages.  Individual files from the {mupdf,leptonica,tesseract}_mod/
+# directories are intended to replace the corresponding source files in the
+# upstream packages, for a particular version of that upstream package.
+#
+# The previous packaging approach involved manually replacing these files and
+# producing patches which could then be applied to those packages.  The benefit
+# of creating and applying these patches in this way is that minor updates
+# (esp. security fixes) might still allow these patches to apply successfully.
+#
+# The current approach creates these patches using nix. This has the benefit of
+# easier review and should hopefully be simpler to update in the future.
+
+let
+  # Create a patch against src based on changes applied in patchCommands
```

Comment:
That actually looks so useful that it might be nice to have in the standard library :)

Reply:
Yeah, it does seem pretty generic. I wonder if there are other places in nixpkgs that could use something like this.